crates/ruff_dev/src/generate_rules_table.rs

```diff
@@ -69,15 +83,28 @@ pub(crate) fn generate() -> String {
     // Generate the table string.
     let mut table_out = String::new();
 
-    table_out.push_str(&format!(
-        "The {FIX_SYMBOL} emoji indicates that a rule is automatically fixable by the `--fix` command-line option."));
-    table_out.push('\n');
+    table_out.push_str("### Legend");
     table_out.push('\n');
 
     table_out.push_str(&format!(
-        "The {PREVIEW_SYMBOL} emoji indicates that a rule is in [\"preview\"](faq.md#what-is-preview)."
+        "{SPACER}{STABLE_SYMBOL}{SPACER} The rule is stable."
     ));
-    table_out.push('\n');
+    table_out.push_str("<br />");
+
+    table_out.push_str(&format!(
+        "{SPACER}{PREVIEW_SYMBOL}{SPACER} The rule is unstable and is in [\"preview\"](faq.md#what-is-preview)."
+    ));
+    table_out.push_str("<br />");
+
+    table_out.push_str(&format!(
+        "{SPACER}{WARNING_SYMBOL}{SPACER} The rule has been deprecated and will be removed in a future release."
+    ));
+    table_out.push_str("<br />");
+
+    table_out.push_str(&format!(
+        "{SPACER}{FIX_SYMBOL}{SPACER} The rule is automatically fixable by the `--fix` command-line option."
+    ));
+    table_out.push_str("<br />");
```

Comment:
I kind of preferred what we had before over a separate section, but I assume you tried to just extend the previous layout and this looked better?

Reply:
For the legend specifically? I didn't like how much blank space there was so I tried removing that and it felt awkward. I find having all of sentences weirdly verbose for documenting the meanings of the icons.

One this we could do here is link to the legend from the icons in the rule table. I've tried clicking them multiple times :D

Reply:
That's interesting. We could even put it at the bottom I guess.

Reply:
Let's focus on this separately #9711